File: .API/Cloud/Submission.cs
// Decompiled with JetBrains decompiler
// Type: CloudX.Shared.Submission
// Assembly: CloudX.Shared, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 2635624C-5F24-4EFB-ACD1-7E9C1349B2F5
// Assembly location: F:\SteamLibrary\steamapps\common\NeosVR\HeadlessClient\CloudX.Shared.dll

using Newtonsoft.Json;
using System;
using System.Text.Json.Serialization;

namespace CloudX.Shared
{
  [Serializable]
  public class Submission
  {
    [JsonProperty(PropertyName = "id")]
    [JsonPropertyName("id")]
    public string Id { get; set; }

    [JsonProperty(PropertyName = "ownerId")]
    [JsonPropertyName("ownerId")]
    public string GroupId { get; set; }

    [JsonProperty(PropertyName = "targetRecordId")]
    [JsonPropertyName("targetRecordId")]
    public RecordId TargetRecordId { get; set; } = new RecordId();

    [JsonProperty(PropertyName = "submissionTime")]
    [JsonPropertyName("submissionTime")]
    public DateTime SubmissionTime { get; set; }

    [JsonProperty(PropertyName = "submittedById")]
    [JsonPropertyName("submittedById")]
    public string SubmittedById { get; set; }

    [JsonProperty(PropertyName = "submittedByName")]
    [JsonPropertyName("submittedByName")]
    public string SubmittedByName { get; set; }

    [JsonProperty(PropertyName = "featured")]
    [JsonPropertyName("featured")]
    public bool Featured { get; set; }

    [JsonProperty(PropertyName = "featuredByUserId")]
    [JsonPropertyName("featuredByUserId")]
    public string FeaturedByUserId { get; set; }

    [JsonProperty(PropertyName = "featuredTimestamp")]
    [JsonPropertyName("featuredTimestamp")]
    public DateTime? FeaturedTimestamp { get; set; }
  }
}

File: src/api/Submission.ts
import { RecordId } from "./RecordId";
import type { RecordIdJSON } from "./RecordId";
export class Submission {
	public Id: string;
	public GroupId: string;
	public TargetRecordId: RecordId;
	public SubmissionTime: Date;
	public SubmittedById: string;
	public SubmittedByName: string;
	public Featured: boolean;
	public FeaturedByUserId: string;
	public FeaturedTimestamp?: Date;
	constructor($b: SubmissionJSON) {
		this.Id = $b.id;
		this.GroupId = $b.ownerId;
		this.TargetRecordId =
			$b.targetRecordId instanceof RecordId
				? $b.targetRecordId
				: new RecordId($b.targetRecordId);
		this.SubmissionTime = $b.submissionTime;
		this.SubmittedById = $b.submittedById;
		this.SubmittedByName = $b.submittedByName;
		this.Featured = $b.featured;
		this.FeaturedByUserId = $b.featuredByUserId;
		this.FeaturedTimestamp = $b.featuredTimestamp;
	}
	toJSON(): SubmissionJSON {
		return {
			id: this.Id,
			ownerId: this.GroupId,
			targetRecordId: this.TargetRecordId.toJSON(),
			submissionTime: this.SubmissionTime,
			submittedById: this.SubmittedById,
			submittedByName: this.SubmittedByName,
			featured: this.Featured,
			featuredByUserId: this.FeaturedByUserId,
			featuredTimestamp: this.FeaturedTimestamp,
		};
	}
}
export interface SubmissionJSON {
	id: string;
	ownerId: string;
	targetRecordId: RecordIdJSON;
	submissionTime: Date;
	submittedById: string;
	submittedByName: string;
	featured: boolean;
	featuredByUserId: string;
	featuredTimestamp?: Date;
}
